errors.go

```diff
@@ -6,74 +6,93 @@ import (
 	"reflect"
 )
 
-type ErrContext interface {
-	message() string
+type ErrAuthFailed struct {
+	Remote bool
+	Inner  error
 }
-type Error[C ErrContext] struct {
-	Inner   error
-	Context C
+
+func (e ErrAuthFailed) Error() string {
+	var msg string
+	if e.Remote {
+		msg = "authentication failed"
+	} else {
+		msg = "failed to send authentication request"
+	}
+
+	return fmt.Sprintf("%s: %v", msg, e.Inner)
 }
 
-func (e Error[C]) Unwrap() error {
+func (e ErrAuthFailed) Unwrap() error {
 	return e.Inner
 }
 
-func (e Error[C]) Error() string {
-	msg := e.Context.message()
-	if e.Inner != nil {
-		return fmt.Sprintf("%s: %v", msg, e.Inner.Error())
-	} else {
-		return msg
-	}
+func (e ErrAuthFailed) Is(target error) bool {
+	return reflect.TypeOf(e) == reflect.TypeOf(target)
```

Comment:
Since this is a concrete type, can't we do:

```
_, ok := target.(ErrAuthFailed)
return ok
```

instead?

Since the concrete type is known at compile-time, it seems like the checked type-assertion should be functionally the same, but faster.

Same comment for the other places using `reflect.TypeOf`

Reply:
Ope, right you are. Will fix